### PR TITLE
Make Spark Max built-in encoder sampling and measurement period configurable

### PR DESCRIPTION
**Pull request by calcmogul**
The number of samples must be between 8 and 64 inclusive.

### CODE REVIEW

File: sysid-application/src/main/native/cpp/view/Generator.cpp [hunk 1]
```diff
@@ -459,17 +460,23 @@ void Generator::Display() {
...
         m_settings.encoderType == sysid::encoder::kCANcoderPro)) {
     // Samples Per Average Setting
     ImGui::SetNextItemWidth(ImGui::GetFontSize() * 2);
-    ImGui::InputInt("Samples Per Average", &m_settings.numSamples, 0, 0);
+    if (ImGui::InputInt("Samples Per Average", &m_settings.numSamples, 0, 0)) {
+      // Spark Max built-in encoder number of samples must be within 8 to 64
+      // inclusive
+      if (mainMotorController == sysid::motorcontroller::kSPARKMAXBrushless &&
+          m_settings.encoderType == sysid::encoder::kSMaxEncoderPort) {
+        m_settings.numSamples = std::clamp(m_settings.numSamples, 8, 64);
```
Comment (Starlight220) [inline sysid-application/src/main/native/cpp/view/Generator.cpp:477]: Do we want to clamp or reject out-of-bounds here?

File: sysid-application/src/main/native/cpp/view/Generator.cpp [hunk 2]
```diff
@@ -459,17 +460,23 @@ void Generator::Display() {
...
+      }
+    }
     CreateTooltip(
         "This helps reduce encoder noise by averaging collected samples "
         "together. A value from 5-10 is reccomended for encoders with high "
```
Comment (Starlight220) [inline sysid-application/src/main/native/cpp/view/Generator.cpp:482]: Does this recommendation tooltip need modification?

File: sysid-application/src/main/native/cpp/view/Generator.cpp (hunk 1 above)
Comment (calcmogul) [reply sysid-application/src/main/native/cpp/view/Generator.cpp:477]: As a rule, we clamp invalid inputs. To restore the original setting, we'd have to store what the previous value was.

**Comment by rzblue** [comment]
Oops, that restriction applies to the measurement period. The depth has a different requirement

**Comment by calcmogul** [comment]
I pushed update UI configuration for the number of samples and measurement period, but I don't know if it's correct. The NEO built-in encoder was never exposed, as far as I can tell, since "built-in" wasn't in the list of encoders. Is that what "encoder port" is supposed to do?

**Pull request merged**